greeter/README.md

```diff
@@ -1,21 +1,706 @@
+
+   ```sh
+   $ lb4 interceptor
+   ? Interceptor name: caching
+   ? Is it a global interceptor? Yes
+
+   Global interceptors are sorted by the order of an array of group names bound to ContextBindings.GLOBAL_INTERCEPTOR_ORDERED_GROUPS. See https://loopback.io/doc/en/lb4/Interceptors.html#order-of-invocation-for-interceptors.
+
+   ? Group name for the global interceptor: ('')
+   create src/interceptors/caching.interceptor.ts
+   update src/interceptors/index.ts
+
+   Interceptor Caching was created in src/interceptors/
+   ```
+
+2. In `src/interceptors/caching.interceptors.ts`, modify the constructor:
+
+   ```ts
+   constructor(
+       @inject(CACHING_SERVICE) private cachingService: CachingService,
+   ) {}
+   ```
+
+   Add and modify the following import statements:
+
+   ```ts
+   import {
+     /* inject, */
+     globalInterceptor,
+     Interceptor,
+     InvocationContext,
+     InvocationResult,
+     Provider,
+     inject,
+     ValueOrPromise,
+   } from '@loopback/context';
+   import {CachingService} from '../services/caching.service';
+   import {CACHING_SERVICE} from '../keys';
+   import {RestBindings} from '@loopback/rest';
+   ```
+
+3. Add the pre-invocation and post-invocation logics in the `intercept` function.
+
+   ```ts
+   //we need to exclude two endpoints for the interceptor:
+   const EXCLUDED_PATHS = ['/explorer/', '/explorer/openapi.json'];
+   // And then skip them in the intercept function
+   //Otherwise when hit the explorer endpoint, it's response also get cached but no timestamp in it, which breaks the cache interceptor.
+   // Add pre-invocation logic here
+   const httpReq = await invocationCtx.get(RestBindings.Http.REQUEST, {
+     optional: true,
+   });
+   /* istanbul ignore if */
+   // A workaround to skip the explorer
+   if (EXCLUDED_PATHS.includes(httpReq.path)) {
+     return next();
+   }
+   const key = httpReq.path;
+   const lang = httpReq.acceptsLanguages(['en', 'zh']) || 'en';
+   const cachingKey = `${lang}:${key}`;
+   const cachedResult = await this.cachingService.get(cachingKey);
+   if (cachedResult) {
+     console.error('Cache found for %s %j', cachingKey, cachedResult);
+     return cachedResult;
+   }
+
+   const result = await next();
+
+   // Add post-invocation logic here
+   await this.cachingService.set(cachingKey, result);
+   return result;
+   ```
+
+4. Then bind Observer and Interceptor to you app
+   in `src/application.ts` file, add these lines:
+
+   ```ts
+   this.bind(RestExplorerBindings.CONFIG).to({
+     path: '/explorer',
+   });
+   // bind interceptor and service to app
+   this.add(createBindingFromClass(CachingService, {key: CACHING_SERVICE}));
+   this.add(createBindingFromClass(CachingInterceptor));
+
+   this.component(RestExplorerComponent);
+   this.component(GreetingComponent);
+   ```
+
+### Step 4: Try it out again!
+
+Let's try it out to see the caching server and the observer that we just bind. Restart your application.
```

Comment:
Let's try it out to see the caching `service` and the observer that we just `bound` to the application. Restart the application.